Userland/Libraries/LibSQL/ResultSet.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022, Jan de Visser <jan@de-visser.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibSQL/ResultSet.h>
+
+namespace SQL {
+
+size_t ResultSet::binary_search(Tuple const& sort_key, size_t low, size_t high)
```

Comment:
We have a generic binary search implementation at `AK/BinarySearch.h`, does that work here instead of new implementation?

Reply:
I tried it, but it doesn't seem to return to always return proper index to insert at if it can't find the key it's looking for. It's an off-by-one, but I don't want to touch it because that would probably break stuff everywhere.

I'll have a look at it later though, because this should work.